portfolio/src/main/webapp/script.js

```diff
@@ -85,13 +86,61 @@ function makeComment(heading, content) {
 function getDropdownVal() {
     const dropdown = document.getElementById("num-comments");
     dropdown.onchange = changeDropdownVal;
-    if (localStorage["num-comments"]) {
-        dropdown.value = localStorage["num-comments"];
+    if (sessionStorage["num-comments"]) {
+        dropdown.value = sessionStorage["num-comments"];
     }   
 }
 
 function changeDropdownVal() {
     const dropdown = document.getElementById("num-comments");
-    localStorage["num-comments"] = dropdown.value;
+    sessionStorage["num-comments"] = dropdown.value;
     this.form.submit();
+}
+
+function deleteMessages() {
+    console.log("Fetching response.");
+    const responsePromise = fetch("/delete-data", {method:"POST"});
+    responsePromise.then(handleDelete);
+}
+
+function handleDelete(response) {
+    console.log("Handling the delete response.");
+    const deletePromise = response.text();
+    deletePromise.then(deleteTextFromDom);
+}
+
+function deleteTextFromDom(text) {
+    console.log("Deleting text from DOM: "+text);
+    const table = document.getElementById("comment-section");
+
+    var i;
+    for (i = 0; i < table.children.length; i++) {
+        const comment = table.children[i];
+        comment.remove();
+    }
+}
+
+function setCurrentPage() {
+    const current = document.getElementById("current-page");
+    if (sessionStorage["current-page"]) {
+        current.innerText = sessionStorage["current-page"];
+    } else {
+        sessionStorage["current-page"] = 1;
+    }
+}
+
+function changeButtonValUp() {
+    const pagination = document.getElementById("pag");
+    pagination.value = 1;
+    sessionStorage["current-page"]++;
+    setCurrentPage();
+}
```

Comment:
What if we are already at the max page. Like if the server doesn't have any more comments. Double check that the server won't break if the requested page number is too high.

Reply:
It doesn't break, it just shows a blank page of comments. I could figure out a way to make the comments not go to the next page if there are no comments, but I wouldn't know how to not increase the page number in the js, if that makes sense.

Reply:
I think you could prevent it from increasing the same way you prevent the page from decreasing when you are decreasing the page. if the get request sends the number of total comments, you can do math to calculate the total number of pages that should be available given the number of comments per page. But that can all be done in a separate PR. i just wanted to make sure this didn't crash the server or anything.